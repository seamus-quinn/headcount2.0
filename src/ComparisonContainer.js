import React from 'react';
import Card from './Card';
import ComparisonCard from './ComparisonCard';
import './styles/ComparisonContainer.css';
import PropTypes from 'prop-types';

const ComparisonContainer = ({
  comparedSchools,
  schoolData,
  schoolRepository,
  updateComparedSchools
}) => {
  const cardOneData = schoolData[comparedSchools[0]];
  const cardTwoData = schoolData[comparedSchools[1]];

  return (
    <div className="compare-wrapper">
      <Card
        cardData={cardOneData}
        updateComparedSchools={updateComparedSchools}
        className="card border"
      />
      {comparedSchools.length === 2 &&
        <div>
          <ComparisonCard
            schoolRepository={schoolRepository}
            comparedSchools={comparedSchools}
          />
          <Card
            cardData={cardTwoData}
            updateComparedSchools={updateComparedSchools}
            className="card border"
          />
        </div>}
    </div>
  );
};

ComparisonContainer.propTypes = {
  comparedSchools: PropTypes.array,
  schoolData: PropTypes.object,
  schoolRepository: PropTypes.object,
  updateComparedSchools: PropTypes.func
};

export default ComparisonContainer;
