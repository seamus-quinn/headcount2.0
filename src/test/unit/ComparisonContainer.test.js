import React from 'react';
import { shallow } from 'enzyme';
import ComparisonContainer from '../../ComparisonContainer';

describe('ComparisonContainer', () => {
  let comparisonContainer;
  let mockComparedSchools = ['COLORADO', 'ACADEMY 20'];
  let mockSchoolData = {
    'COLORADO':
      {
        location: 'COLORADO',
        stats: { 2003: .021 }
      }
  };
  let mockSchoolRepository;
  let mockUpdateComparedSchools;

  beforeEach(() => {
    comparisonContainer = shallow(<ComparisonContainer
      comparedSchools={mockComparedSchools}
      schoolData={mockSchoolData}
      schoolRepository={mockSchoolRepository}
      updateComparedSchools={mockUpdateComparedSchools}
    />);
  });

  it('matches snapshot', () => {
    expect(comparisonContainer).toMatchSnapshot();
  });

});
