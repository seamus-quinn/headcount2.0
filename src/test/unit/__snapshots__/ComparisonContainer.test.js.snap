// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonContainer matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonContainer
    comparedSchools={
        Array [
            "COLORADO",
            "ACADEMY 20",
          ]
    }
    schoolData={
        Object {
            "COLORADO": Object {
              "location": "COLORADO",
              "stats": Object {
                "2003": 0.021,
              },
            },
          }
    }
    schoolRepository={undefined}
    updateComparedSchools={undefined}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          cardData={
                    Object {
                              "location": "COLORADO",
                              "stats": Object {
                                "2003": 0.021,
                              },
                            }
          }
          className="card border"
          updateComparedSchools={undefined}
/>,
        <div>
          <ComparisonCard
                    comparedSchools={
                              Array [
                                        "COLORADO",
                                        "ACADEMY 20",
                                      ]
                    }
                    schoolRepository={undefined}
          />
          <Card
                    cardData={undefined}
                    className="card border"
                    updateComparedSchools={undefined}
          />
</div>,
      ],
      "className": "compare-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "cardData": Object {
            "location": "COLORADO",
            "stats": Object {
              "2003": 0.021,
            },
          },
          "className": "card border",
          "updateComparedSchools": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ComparisonCard
              comparedSchools={
                            Array [
                                          "COLORADO",
                                          "ACADEMY 20",
                                        ]
              }
              schoolRepository={undefined}
/>,
            <Card
              cardData={undefined}
              className="card border"
              updateComparedSchools={undefined}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "comparedSchools": Array [
                "COLORADO",
                "ACADEMY 20",
              ],
              "schoolRepository": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "cardData": undefined,
              "className": "card border",
              "updateComparedSchools": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            cardData={
                        Object {
                                    "location": "COLORADO",
                                    "stats": Object {
                                      "2003": 0.021,
                                    },
                                  }
            }
            className="card border"
            updateComparedSchools={undefined}
/>,
          <div>
            <ComparisonCard
                        comparedSchools={
                                    Array [
                                                "COLORADO",
                                                "ACADEMY 20",
                                              ]
                        }
                        schoolRepository={undefined}
            />
            <Card
                        cardData={undefined}
                        className="card border"
                        updateComparedSchools={undefined}
            />
</div>,
        ],
        "className": "compare-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "location": "COLORADO",
              "stats": Object {
                "2003": 0.021,
              },
            },
            "className": "card border",
            "updateComparedSchools": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ComparisonCard
                comparedSchools={
                                Array [
                                                "COLORADO",
                                                "ACADEMY 20",
                                              ]
                }
                schoolRepository={undefined}
/>,
              <Card
                cardData={undefined}
                className="card border"
                updateComparedSchools={undefined}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "comparedSchools": Array [
                  "COLORADO",
                  "ACADEMY 20",
                ],
                "schoolRepository": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "cardData": undefined,
                "className": "card border",
                "updateComparedSchools": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
